cli: add completion support for fish shell

sayboras opened the pull request:
<!-- Description of change -->

Fixes: https://github.com/cilium/cilium/issues/11283

![image](https://user-images.githubusercontent.com/9019229/80809662-755d5f00-8c05-11ea-8285-84c6367a8ceb.png)

```release-note
Add completion support for fish shell
```

sayboras committed:
cli: Add completion support for fish shell

Recently, fish completion is supported in cobra v1.0.0.
It would be nice to leverage this feature in cilium.feature/cilium-fish

Fixes: #11283
Signed-off-by: Tam Mach <sayboras@yahoo.com>

qmonnet commented on cilium/cmd/root.go:
```diff
@@ -62,7 +62,7 @@ func init() {
 	viper.BindPFlags(flags)
 	rootCmd.AddCommand(newCmdCompletion(os.Stdout))
-	rootCmd.SetOut(os.Stderr)
+	rootCmd.SetOut(os.Stdout)
```
Is this change intentional? I wonder if it might be the cause for the CI failure?

sayboras replied:
This change is required as internally _cilium __complete_ command will be called, and it's using the stdout for showing completion options. 

I just checked the failure, seems like we are expecting help output in stderr. Let me see if there is easy way to fix this.

sayboras committed:
test: Update the cli gingko test

This commit is to update the cli test cases for help command

Originally, help command output is printing to stderr instead of stdout.
This commit might be reverted if it's not agreed by maintainer

Signed-off-by: Tam Mach <sayboras@yahoo.com>
